test/Network/Socket.cpp

```diff
@@ -0,0 +1,17 @@
+#include <SFML/Network/Socket.hpp>
+
+#include <doctest/doctest.h>
+
+#include <type_traits>
+
+TEST_CASE("sf::Socket class - [network]")
+{
+    SUBCASE("Copy and move semantics")
+    {
+        static_assert(!std::is_constructible<sf::Socket>::value);
```

Comment:
I'm guessing that only protected / private constructors are tested, and not stuff like abstract classes not being constructible?

Reply:
It would make sense for all non-constructible classes to be tested but I suppose I just forgot to add that assertions to some types.